flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliTableauResultView.java

```diff
@@ -38,31 +38,46 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
 
 /** Print result in tableau mode. */
 public class CliTableauResultView implements AutoCloseable {
 
+    public static final long DEFAULT_QUERY_BEGIN_TIME = -1L;
+
     private final Terminal terminal;
     private final ResultDescriptor resultDescriptor;
 
     private final ChangelogResult collectResult;
     private final ExecutorService displayResultExecutorService;
 
-    public CliTableauResultView(final Terminal terminal, final ResultDescriptor resultDescriptor) {
-        this(terminal, resultDescriptor, resultDescriptor.createResult());
+    private final long queryBeginTime;
+
+    public CliTableauResultView(
+            final Terminal terminal, final ResultDescriptor resultDescriptor, long queryBeginTime) {
+        this(terminal, resultDescriptor, resultDescriptor.createResult(), queryBeginTime);
     }
 
     @VisibleForTesting
     public CliTableauResultView(
             final Terminal terminal,
             final ResultDescriptor resultDescriptor,
             final ChangelogResult collectResult) {
+        this(terminal, resultDescriptor, collectResult, DEFAULT_QUERY_BEGIN_TIME);
```

Comment:
Do we need to introduce a static member fields here if it is just used by tests? I think we can use the System.currentMills() if no one specifies the begin time.

Reply:
the static member field will be used to control the time cost display logic